src/python/pants/binaries/binary_tool.py

```diff
@@ -70,10 +84,35 @@ def select(self, context=None):
         version = old_opts.get(self.replaces_name)
     return self._select_for_version(version)
 
+  @memoized_property
+  def _binary_util(self):
+    return BinaryUtil.Factory.create()
+
+  # can override this and call super() to compose
+  @classmethod
+  def support_dir_paths(cls):
```

Comment:
Returning path segments seems overly complicated and is confusing.

How about we just eliminate the whole support dir concept entirely, and just do a one-time rearrangement of our binary tools so that the role of support dir is fulfilled by the binary name.

E.g., why do we need to have `protoc` under `bin/protobuf` when we can have it under `bin/protoc` ? And we can get rid of the `script` subdir entirely, and put everything under `bin`. There's no need to distinguish between them via path prefix.

10 minutes of S3 copying could simplify this code a lot.

Reply:
I agree that it is confusing, I spent a lot of time thinking about the right thing to do here and I wasn't super satisfied with this, but I didn't know we could do what you described. I don't know how easy modifying S3 to make this work would be, but I agree otherwise that sticking everything in the `bin/` directory is a great idea and would remove many lines of code here.

Reply:
Should probably run this by @stuhood and @jsirois, but I think it should be fine to just change the structure in our S3 dirs. We'd keep the old structure to support existing Pants versions, of course. I have relevant AWS permissions, so I can take a crack at it.

Reply:
Additionally, requiring the `options_scope` to be the same as the directory the binary is stored under is what I really wanted to do, but I also thought that was off limits, so if we can rearrange the structure of pants binaries easily enough that would also reduce some unnecessary complexity.

Reply:
That's a little less robust, as the option scope can change via a deprecation cycle, and the deprecator may not think to modify S3. Whereas the binary name is fixed.

In some cases we can't have the options_scope be the same as the name for legacy reasons (e.g., the scope of `buildozer` is `buildozer-binary` because `buildozer` is the scope of the task that runs the buildozer binary).

Reply:
That makes perfect sense -- I didn't see that comment in time! I'll add back the ability to customize the tool's containing directory separately from `options_scope`.

Reply:
I think the directory should just be `bin/<tool_name>`, e.g., `bin/protoc`.

Reply:
In most cases they already are, I bet...

Reply:
OK, it turns out things are a little more complicated - in some cases the "binary name" for the purpose of fetching is, e.g., `go.tar.gz`.  But I think your change will already do the right thing here, ya?

Reply:
And we can get rid of the repetition of archive unpacking code in go_distribution.py and elsewhere.

Reply:
OK, I did the relevant S3 copying - really just moving the `scripts/` to `bin/`, and copying protobuf->protoc, everything else under `bin/` was already in the appropriately named dir.

So now you can hardwire the supportdir BinaryTool passes to BinaryUtils to `bin/<toolname>` and it should work!